rust/src/dcerpc/dcerpc.rs

```diff
@@ -0,0 +1,2600 @@
+        };
+        let rpc_vers = match self.get_hdr_rpc_vers() {
+            Some(x) => x,
+            None => 0,
+        };
+        let fraglenleft = self.fraglenleft;
+
+        // Update the stub params based on the packaet type
+        match self.get_hdr_pkt_type() {
+            Some(x) => match x {
+                DCERPC_TYPE_REQUEST => {
+                    if let Some(ref mut req) = self.request {
+                        retval = evaluate_stub_params(
+                            input,
+                            input_len,
+                            hdrflags1,
+                            fraglenleft,
+                            &mut req.stub_data_buffer,
+                            &mut req.stub_data_buffer_len,
+                            &mut req.stub_data_buffer_reset,
+                            rpc_vers,
+                        );
+                    }
+                }
+                DCERPC_TYPE_RESPONSE => {
+                    if let Some(ref mut resp) = self.response {
+                        retval = evaluate_stub_params(
+                            input,
+                            input_len,
+                            hdrflags1,
+                            fraglenleft,
+                            &mut resp.stub_data_buffer,
+                            &mut resp.stub_data_buffer_len,
+                            &mut resp.stub_data_buffer_reset,
+                            rpc_vers,
+                        );
+                    }
+                }
+                _ => {
+                    SCLogDebug!("Unrecognized packet type");
+                    return 0;
+                }
+            },
+            None => {
+                return 0;
+            }
+        }
+        // Update the remaining fragment length
+        self.fraglenleft -= retval;
+
+        retval
+    }
+
+    pub fn process_header(&mut self, input: &[u8]) -> i32 {
+        match parser::parse_dcerpc_udp_header(input) {
+            Ok((_leftover_bytes, header)) => {
+                if header.rpc_vers != 4 {
+                    SCLogDebug!("DCERPC UDP Header did not validate.");
+                    return -1;
+                }
+                let mut uuidentry = DCERPCUuidEntry::new();
+                let auuid = header.activityuuid.to_vec();
+                uuidentry.uuid = auuid;
+                self.uuid_list.push(uuidentry);
+                self.header = Some(header);
+                80
+            }
+            Err(nom::Err::Incomplete(_)) => {
+                // Insufficient data.
+                SCLogDebug!("Insufficient data while parsing DCERPC request");
+                -1
+            }
+            Err(_) => {
+                // Error, probably malformed data.
+                SCLogDebug!("An error occurred while parsing DCERPC request");
+                -1
+            }
+        }
+    }
+
+    pub fn handle_input_data(&mut self, input: &[u8]) -> AppLayerResult {
```

Comment:
got confused a bit as I see this method twice, once in the tcp and once in the udp handling. Would it be feasible to split those 2 implementations into 2 files?